### ISSUE TITLE
GP02 and X1 support

### ISSUE THREAD

**Issue by Dataweaver42** [Perdition-117/COM3D2.DressCode#6]
GP02 adds a "Private Mode" where the Office mode is replaced by a potentially random location featuring your wife. This could be a problem, as the DressCode button is specific to the Office mode; it means that once I have a wife and turn on Private Mode, I'll lose access to DressCode; switching back to Office mode whenever I want to edit the dress code is doable, but it will be a significant inconvenience. 

On the other hand, this also suggests another possible kind of "event" that DressCode could take advantage of: Private Mode. I would love to have my wife appear in more casual attire rather than a maid uniform when she's hanging around in Private Mode. Assuming that that can be done, X1 extends this further with a series of "honeymoon" Events; if there's a way to identify when a particular Event is X1-specific, it would be nice to apply the "wife outfit" to those scenes as well. That said, I fully understand if this turns out to be too big of a headache to implement; I mention it only because you mentioned that you'd like to expand the range of scenes that DressCode covers.

**Comment by Perdition-117** [comment]
Missing private mode button should hopefully be fixable. I do have GP-02, but no waifu!

I don't know how private mode works, but crucially there needs to exist an opportunity to restore your regular attire after the scene is over. So if you can, say, save in private mode, that's going to be a problem. Depending on how private mode works however, there may be a way around that. Would this "scene" only encompass the "office" screen, or does the waifu appear anywhere else? (ie is that all private mode does?)

I intend to get X1 at some point, but for now I don't really have an idea of how it works. All I know is there's a new screen for it. Are you referring to just regular events that take place during honeymoon? And for the same private mode waifu costume to apply in these events? I could possibly imagine a dedicated honeymoon costume, even.

In general, I'm always open to suggestions, and I often have very ambitious plans myself, but what I want does not always align with what efforts I'm willing to make, and I also have countless of projects (including playing games) among which I try to split my limited time. :)

In the case of DressCode, to be honest, the part I dread the most is making the necessary UI changes. I'm happy with the UI I have now, but getting there was a pain and mostly possible thanks to piecing together parts of other systems. That's not to say I don't *want* to change it, just that doing so would be difficult, and I don't want to settle for suboptimal solutions. The logical bits are probably easy in comparison.

That last paragraph is probably more for your ticket regarding expanded location support, actually. Adding UI support for another top level scene along side dance, pole dance and yotogi is easy enough.

P.S. Sorry for the walls of text!

**Comment by Dataweaver42** [comment]
Private Mode comes in two forms: the first is a straight-up replacement for Office Mode; and the only thing it does is to change the location being viewed and to put your wife in that location for your viewing pleasure.  If possible, there should be a DressCode button here. The real difficulty involves getting back to the right place: if the player has Private Mode turned on, then you don't want DressCode to return them to the Office screen. I wouldn't worry about the results of editing, though: as things stand with Private Mode, if you go to another screen such as Maid Admin, Events, or Facilities, and then return to Private Mode, the location and maid displayed in Private Mode gets reloaded from scratch. So if someone goes to DressCode and returns to Private Mode, the costume changes should take effect immediately.  

The other part of Private Mode is that you can click on your wife to go to a screen with the same background and maid, but with a menu of location-specific events to peruse. I mention it only to point out that DressCode wouldn't need to mess with that part at all: I wouldn't expect to find a DressCode button there any more than I'd expect to find it on the Events screen. 

X1 doesn't have a new screen; it has several new locations which the X1-specific Events make use of, and which can also be used for Yotogi; but for the purpose of DressCode, the only time I'd expect a marriage-specific outfit to be needed would be during the X1 Events. And while it would be nice to have events taking place at the beach to use a different outfit than events taking place at the Love Hotel or the "marriage resort" (specifically to promote the use of swimwear costumes), I'm not actually asking for that; a single marriage-themed outfit per maid for all X1 events would suffice.

**Comment by Perdition-117** [comment]
So the first form of private mode is just the office screen with a different background? What I mean is, how DressCode works is whenever a dress code scene loads, your maid's *permanent* costume is changed. I therefore need to explicitly restore the original costume (stored in memory) when the scene is over. If you can save and quit the game while in private mode, that means the dress code costume will be your regular costume next time you load the game. As I said though, there may be a workaround.

I will have to try both of these myself regardless before I can do anything.

So, to summarise, am I right?

- Private mode office scene
- X1 event scene

**Comment by Dataweaver42** [comment]
Pretty much, yes.

**Comment by Dataweaver42** [comment]
> What I mean is, how DressCode works is whenever a dress code scene loads, your maid's permanent costume is changed. 
Oh? Remember that I mentioned elsewhere that the Yotogi system gives you the option to Change Costume while you're choosing what activities to perform? I decided to try that out while a DressCode modifiation was in effect, in order to see what would happen.  What I found was this: 
![img20230316224305](https://user-images.githubusercontent.com/21045510/225797506-7e2832c6-b8cc-47a7-a4bd-b6262f9dc522.png)
This was what I saw as soon as the Edit screen loaded. Notice that there's no correlation whatsoever between what she's shown to be wearing and what the controls on the right _say_ that she's wearing. As soon as I attempted to change something, her appearance immediately changed to conform to what's on the right, and I continued the Edit session with the original costume she had been wearing before the Yotogi session had begun. When I finished the edit, the one-piece swimsuit that DressCode had put her in immediately reasserted itself for the rest of the Yotogi session, including a brief scene that took place afterward; but when the session ended, the results of the edit that I did during the session snapped back into effect. I don't see how that squares with the costume permanently changing before the session begins and having to be put back the way it was when the scene is through.

**Comment by Dataweaver42** [comment]
Update: my concerns about DressCode possibly not being available once you turn on Personal Mode was unfounded. There's nothing here that needs fixing; all that's left is the feature request. And that's a low-priority item.

**Comment by Perdition-117** [comment]
I had not considered the costume button during yotogi setup. I don't fully understand the behavior you're describing, but I can confirm the costume mismatch upon launching edit mode and offer a brief explanation. What happens at this point is you leave the yotogi scene, restoring the original costume, why the menu shows the original costume. The yotogi costume remains, I think, because the game has not triggered a costume update on the maid object (which would not be necessary were it not for DressCode), so that's the actual bug here. Of course, the expected behavior would probably be to launch edit mode for yotogi dress code, so I'll look into that.
